### PR TITLE
Close #376 semanticId ExternalReferences and Close #366 individual versioning

### LINKED ISSUE #366: Individual versioning of metamodel element IDs
**Is your feature request related to a problem? Please describe.**
Up to V3.0 all elements got a new version, e.g. `https://admin-shell.io/aas/3/1/AssetAdministrationShell` even if there was no change.

If there are no changes then the version should not be increased. Inherited class changes need to be considered in the version. So if the class A that is inherited by class B is changed then also the version of class B is increased.
Same for type changes. If a type A is changed and used as type in one of the attributes in class B then the version of class B needs to be increased as well.
The third case is the change of constraints: if a constraint has an impact on the allowed values of a class B the the version of class B needs to be increased.

Same for API operations in Part 2

### LINKED ISSUE #376: HasSemantics/semanticId shall always be an External Reference
**Is your feature request related to a problem? Please describe.**

In V3.0 it is recommended to use an external reference for semanticId.

image

However, there are many questions what exactly this means.

In Workstream AAS Specs on 2024-03-07 the following was decided:

- enhance note and mention that this also holds if there is a ConceptDescription available
- add information that in future there might be a constraint requiring to only use ExternalReference
- add note that if ExternalReference is not used, the lookup of the ConceptDescription may fail

### PR DESCRIPTION
**Pull request by BirgitBoss**
#376 Improvement of note recommending to use external references for HasSemantics/semanticId

#366 reverse individual versioning

### CODE REVIEW

**Comment by mhrimaz** [comment]
@BirgitBoss I am very sorry that I ask my question here. 
If in the future we makes it mandatory that semanticId should be `ExternalReference/GlobalReference`, then why we even have `ModelReference/ConceptDescription`?
@s-heppner This will be closely related to https://github.com/eclipse-basyx/basyx-python-sdk/issues/234 and https://github.com/eclipse-basyx/basyx-python-sdk/issues/244.

In the end, how can I understand that a semanticId like `https://www.hsu-hh.de/aut/aas/physicaladdress` is just a url or a `ConceptDescription` that I can find in some ConceptDescriptionRepository?

**Comment by BirgitBoss** [comment]
There are other attributes with type "Reference" that are not affected. 
In general a data consumer does not know the endpoints where to get concept descriptions. Additional discovery services are needed. This is especially true in decentralized systems.

File: documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_Common.adoc [hunk 1]
```diff
@@ -303,7 +303,10 @@ Identifier of a supplemental semantic definition of the element called supplemen
 
 
 ====
-Note: it is recommended to use an external reference.
+Note: it is recommended to use an external reference. This is even recommended in the case a xref:IDTA-01001_Metamodel_Designators.adoc#ConceptDescription[ConceptDescription] exists. Otherwise lookup of the corresponding concept description may fail.
+
+In future releases a constraint may be added that makes is mandatory to use external references.
```
Comment (s-heppner) [inline documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_Common.adoc:308]:
```suggestion
In future releases a constraint may be added that makes it mandatory to use external references.
```

Small typo

Comment (s-heppner) [reply documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_Common.adoc:308]: What is implied here, but not directly stated, is that we do not want the user to reference a "local" ConceptDescription that for example is shipped with an AASX package, right?

I fear that this formulation will create more AASX packages where constraint AASd-123 is violated: 

> Constraint AASd-123: For model references, i.e. References with Reference/type = ModelReference, the value of Key/type of the first key of Reference/keys shall be one of AasIdentifiables.

The reason for this is that someone who manually creates an AAS sees that one should refer to a `ConceptDescription` with an `ExternalReference` and do so, even if the referenced `ConceptDescription` is located inside the model and therefore needs a `ModelReference`.

**Comment by s-heppner** [comment]
What is implied here, but not directly stated, is that we do not want the user to reference a "local" ConceptDescription that for example is shipped with an AASX package, right?

I fear that this formulation will create more AASX packages where constraint AASd-123 is violated: 

> Constraint AASd-123: For model references, i.e. References with Reference/type = ModelReference, the value of Key/type of the first key of Reference/keys shall be one of AasIdentifiables.

The reason for this is that someone who manually creates an AAS sees that one should refer to a `ConceptDescription` with an `ExternalReference` and do so, even if the referenced `ConceptDescription` is located inside the model and therefore needs a `ModelReference`.

**Comment by BirgitBoss** [comment]
> The reason for this is that someone who manually creates an AAS sees that one should refer to a `ConceptDescription` with an `ExternalReference` and do so, even if the referenced `ConceptDescription` is located inside the model and therefore needs a `ModelReference`.

But we do not state that a concept description shall be referenced. we state that is shall be a concept definition. This is the superterm for any kind of concept definition including concept descriptions which are model elements.

**Comment by s-heppner** [comment]
I get that, however I still think the formulation may lead to confusion, with users potentially utilizing external references to reference model objects. 

Maybe we could clarify the note text with something like this?

```
Note: it is recommended to use an external reference, (e.g. referencing a globally resolvable semantic definition). This...
```

**Comment by BirgitBoss** [comment]
@s-heppner please check again: I incorportated your change proposals

**Comment by BirgitBoss** [comment]
@mjacoby could you also review (added notes in line 309 and line 294, the other changes are just xref corrections)

**Comment by mjacoby** [comment]
@BirgitBoss I do not feel qualified to review this as I have neither participated in this discussion nor am I familiar with AsciiDoc and those xrefs.

**Comment by BirgitBoss** [comment]
> @BirgitBoss I do not feel qualified to review this as I have neither participated in this discussion nor am I familiar with AsciiDoc and those xrefs.

sure, xref need to review, only the note. But I will remove you from the reviewer list, no problem

**Pull request merged**